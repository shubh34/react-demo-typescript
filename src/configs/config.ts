export const dateFormat: string = 'DD MMM YYYY';
export const maxContentCharacterLimit: number = 300;
