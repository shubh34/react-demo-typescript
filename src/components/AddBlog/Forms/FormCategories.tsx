import React, { Component } from 'react';
import { Form, Input, Message } from 'semantic-ui-react';
import { ENTER_KEY, COMMA_KEY } from '../../../configs/config';
import CategoriesContainer from '../../sharedComponents/CategoriesContainer/CategoriesContainer';
interface IFormCategories {
	categories: string[];
	addCategory: Function;
	deleteCategory: Function;
}
interface IState {
	category: string;
}
export class FormCategories extends Component<IFormCategories, IState> {
	constructor(props: IFormCategories) {
		super(props);
		this.state = {
			category: '',
		};
		this.handleChange = this.handleChange.bind(this);
		this.handleDeleteCategory = this.handleDeleteCategory.bind(this);

		this.handleKeyUp = this.handleKeyUp.bind(this);
	}
	handleChange(e: React.ChangeEvent<HTMLInputElement>) {
		e.preventDefault();
		this.setState({
			category: e.target.value,
		});
	}
	handleDeleteCategory(category: string) {
		this.props.deleteCategory(category);
	}
	handleKeyUp(e: any) {
		const key = e.keyCode;

		if ((key === ENTER_KEY || key === COMMA_KEY) && this.state.category.trim()) {
			this.props.addCategory(this.state.category.trim().replace(',', ''));
			this.setState({
				category: '',
			});
		}
	}
	render() {
		const { categories } = this.props;
		return (
			<div className='form-categories'>
				<Form.Field>
					<label>Categories</label>
					{!!categories.length && (
						<CategoriesContainer
							categories={categories}
							hasDeleteIcon
							onCategoryClick={this.handleDeleteCategory}
						/>
					)}
					<Input
						spellCheck
						placeholder='Enter your content'
						onKeyDown={this.handleKeyUp}
						onChange={this.handleChange}
						value={this.state.category}
					/>
					<Message visible info>
						*Please press spacebar or enter to create category.
					</Message>
				</Form.Field>
			</div>
		);
	}
}

export default FormCategories;
