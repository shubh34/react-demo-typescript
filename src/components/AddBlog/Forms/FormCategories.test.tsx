import React from 'react';
import { shallow } from 'enzyme';
import FormCategories from './FormCategories';
import { ENTER_KEY } from '../../../configs/config';

const setup = (setUpProps= {}, container= shallow) => {
    const props={
        categories: ['Travel'],
	    addCategory: jest.fn(),
        deleteCategory: jest.fn(),
        ...setUpProps,
    }
    const wrapper = shallow(<FormCategories {...props}/>);
    return {wrapper, props}
}

describe('FormCategories', () => {
    it('renders without crashing', () => {
        const { wrapper } = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onChange', () => {
        const { wrapper } = setup();
        wrapper.find('Input').simulate('change', {
            target: {
                value: 'test'
            }});
        expect(wrapper.state()).toMatchSnapshot();
    });

    it('should call addCategory on onKeyUp', () => {
        const { wrapper, props } = setup();
        wrapper.find('Input').simulate('change', {
            target: {
                value: 'test'
            }});
        wrapper.update();
        wrapper.find('Input').simulate('keyup', {
                keyCode: ENTER_KEY
            });
       expect(props.addCategory).toBeCalledWith('test');
    });

    it('should not call addCategory on onKeyUp', () => {
        const { wrapper, props } = setup();
        wrapper.find('Input').simulate('change', {
            target: {
                value: 'test'
            }});
            wrapper.find('Input').simulate('keypress', {
                    keyCode: '1'
            });
           expect(props.addCategory).not.toBeCalled();
    });
})

