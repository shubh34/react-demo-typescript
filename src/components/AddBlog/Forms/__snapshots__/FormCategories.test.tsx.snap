// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCategories renders without crashing 1`] = `
<div
  className="form-categories"
>
  <FormField>
    <label>
      Categories
    </label>
    <div>
      <Button
        as="button"
        name="Travel"
      >
        Travel
         
        <Icon
          as="i"
          name="delete"
        />
      </Button>
      <Input
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Enter your content"
        spellCheck={true}
        type="text"
        value=""
      />
      <Message
        info={true}
        visible={true}
      >
        *Please press spacebar or enter to create category.
      </Message>
    </div>
  </FormField>
</div>
`;

exports[`FormCategories should call onChange 1`] = `
Object {
  "category": "test",
}
`;
